src/humanloop.client.ts

```diff
@@ -0,0 +1,175 @@
+    }
+
+    async run(
+        file: File,
+        dataset: Dataset,
+        name?: string,
+        evaluators: Evaluator[] = [],
+        workers: number = 8,
+    ): Promise<EvaluatorCheck[]> {
+        return runEval(this._client, file, dataset, name, evaluators, workers);
+    }
+}
+
+export class HumanloopClient extends BaseHumanloopClient {
+    protected readonly _evaluations: ExtendedEvaluations;
+    protected readonly _prompts_overloaded: Prompts;
+    protected readonly _flows_overloaded: Flows;
+
+    protected readonly OpenAI?: any;
+    protected readonly Anthropic?: any;
+    protected readonly CohereAI?: any;
+
+    protected readonly opentelemetryTracerProvider: NodeTracerProvider;
+    protected readonly opentelemetryTracer: Tracer;
+
+    constructor(
+        _options: BaseHumanloopClient.Options,
+        OpenAI?: any,
+        Anthropic?: any,
+        CohereAI?: any,
+    ) {
+        super(_options);
+
+        this.OpenAI = OpenAI;
+        this.Anthropic = Anthropic;
+        this.CohereAI = CohereAI;
+
+        this._prompts_overloaded = overloadLog(super.prompts);
+
+        this._flows_overloaded = overloadLog(super.flows);
+
+        this._evaluations = new ExtendedEvaluations(_options, this);
+
+        this.opentelemetryTracerProvider = new NodeTracerProvider({
+            spanProcessors: [
+                new HumanloopSpanProcessor(new HumanloopSpanExporter(this)),
+            ],
+        });
+
+        if (OpenAI) {
+            const instrumentor = new OpenAIInstrumentation({
+                enrichTokens: true,
+            });
+            instrumentor.manuallyInstrument(OpenAI);
+            instrumentor.setTracerProvider(this.opentelemetryTracerProvider);
+            instrumentor.enable();
+        }
+
+        if (Anthropic) {
+            const instrumentor = new AnthropicInstrumentation();
+            instrumentor.manuallyInstrument(Anthropic);
+            instrumentor.setTracerProvider(this.opentelemetryTracerProvider);
+            instrumentor.enable();
+        }
+
+        if (CohereAI) {
+            const instrumentor = new CohereInstrumentation();
+            instrumentor.manuallyInstrument(CohereAI);
+            instrumentor.setTracerProvider(this.opentelemetryTracerProvider);
+            instrumentor.enable();
+        }
+
+        this.opentelemetryTracerProvider.register();
+
+        this.opentelemetryTracer =
+            this.opentelemetryTracerProvider.getTracer("humanloop.sdk");
+    }
+
+    // Check if user has passed the LLM provider instrumentors
+    private _assertProviders() {
+        const noProviderInstrumented = [
+            this.OpenAI,
+            this.Anthropic,
+            this.CohereAI,
+        ].every((p) => !p);
+        if (noProviderInstrumented) {
+            throw new Error(
+                "Using Prompt File utility without passing any provider in the " +
+                    "HumanloopClient constructor. Did you forget to pass them?",
```

Comment:
when is this shown to users?

Reply:
ah i see

<img width="1035" alt="image" src="https://github.com/user-attachments/assets/b0aed7e5-8138-4790-a302-1537dc0d1c64" />


i think this is a very passive aggressive thing to tell users.
instead of "did you forget to pass them?", we should say something like "Prompt File utility can only be used when a provider is passed in the Humanloop constructor." or "Humanloop cannot find ... . To ..., pass a provider into the HumanloopClient constructor.".

Also I think the term "Prompt File utility" is very confusing and ambiguous. We should instead specify which function was used. (Unclear why "File" important here; unclear what counts as a "utility" here - and so hard to figure out what's wrong - are all sdk methods under .prompts  "prompt utilities"?;)

Reply:
How about this:
`${func.name}: You must pass at least one LLM client library in the Humanloop client constructor. Otherwise the prompt() utility will not work properly.`

Reply:
The instrumentors require the module and modify the prototype of the client itself. So all client instances would be watched by the OTel instrumentor:
https://github.com/traceloop/openllmetry-js/blob/main/packages/instrumentation-openai/src/instrumentation.ts#L55

Reply:
```
${func.name}: To use the `prompt()` utility, pass in your LLM client library into the Humanloop client constructor; e.g. `HumanloopClient(..., {providerModules: {OpenAI} } )`.
```

clarified; should pass in all relevant llm client libraries.